raiden/network/matrixtransport.py

```diff
@@ -0,0 +1,722 @@
+                    return
+                self._address_to_roomid[peer_address] = room.room_id
+                room.add_listener(self._handle_message)
+            log.debug(
+                'ROOM',
+                room_id=room_id,
+                aliases=room.aliases,
+                listening=should_listen
+            )
+
+    def _handle_invite(self, room_id: str, state: dict):
+        """ Join all invited rooms """
+        room = self._client.join_room(room_id)
+        if not room.canonical_alias:
+            log.warning('Got invited to room without canonical alias - ignoring', room=room)
+            return
+        peer_address = self._get_peer_address_from_room(room.canonical_alias)
+        if not peer_address:
+            log.warning(
+                'Got invited to room we\'re not supposed to be a member of - ignoring',
+                room=room
+            )
+            return
+        self._address_to_roomid[peer_address] = room.room_id
+        room.add_listener(self._handle_message, 'm.room.message')
+        log.debug(
+            'Invited to room',
+            room_id=room_id,
+            aliases=room.aliases
+        )
+
+    def _handle_message(self, room, event):
+        """ Handle text messages sent to listening rooms """
+        if event['type'] != 'm.room.message' or event['content']['msgtype'] != 'm.text':
+            # Ignore non-messages and non-text messages
+            return
+
+        sender_id = event['sender']
+
+        if sender_id == self._client.user_id:
+            # Ignore our own messages
+            return
+
+        user = self._client.get_user(sender_id)
+
+        peer_address = self._userids_to_address.get(sender_id)
+        if not peer_address:
+            try:
+                # recover displayname signature
+                peer_address = signing.recover_address(
+                    sender_id.encode(),
+                    signature=data_decoder(user.get_display_name()),
+                    hasher=eth_sign_sha3
+                )
+            except AssertionError:
+                log.warning('INVALID MESSAGE', sender_id=sender_id)
+                return
+            node_address_hex = address_encoder(peer_address)
+            if node_address_hex.lower() not in sender_id:
+                log.warning(
+                    'INVALID SIGNATURE',
+                    peer_address=node_address_hex,
+                    sender_id=sender_id
+                )
+                return
+            self._userids_to_address[sender_id] = peer_address
+
+        data = event['content']['body']
+        if data.startswith('0x'):
+            message = message_from_bytes(data_decoder(data))
+        else:
+            message_dict = json.loads(data)
+            log.trace('MESSAGE_DATA', data=message_dict)
+            message = message_from_dict(message_dict)
+
+        if isinstance(message, SignedMessage) and not message.sender:
+            # FIXME: This can't be right
```

Comment:
also same question here.

Reply:
I'm wondering why the message.sender field isn't encoded into the json representation and therefore has to be set by the transport in the receive method. It works as is for now but it should be discussed how to do it nicer in the future.